### ISSUE TITLE
Incorrect results with Access protocol and Nx.argsort(x, direction: :desc)

### ISSUE THREAD

**Issue by msluszniak** [elixir-nx/nx#1246]
The following code
```elixir
y_score = Nx.tensor([1, 0, 1, 1, 0])
desc_ind = Nx.argsort(y_score, direction: :desc)
y_score = Nx.take(y_score, desc_ind)
y_score[[1..-1]]
```
returns 
```
#Nx.Tensor<
  s64[4]
  [0, 0, 1, 1]
>
```
while when we replace last line with
```elixir
Nx.take(y_score, Nx.tensor([1, 2, 3, 4]))
````

we got
```
#Nx.Tensor<
  s64[4]
  [1, 1, 0, 0]
>
```

**Comment by polvalente** [comment]
You should use take_along_axis to apply indices provided by argsort

**Comment by polvalente** [comment]
Also, please provide all intermediate values so we can see any possible errors.

It's better if you avoid variable name rebinding as well

**Comment by msluszniak** [comment]
> You should use take_along_axis to apply indices provided by argsort

Ok, but there is no information about it in the documentation, so it might be confusing.

**Comment by polvalente** [comment]
There's an example in the take_along_axis docs. Perhaps we should move it to argsort

**Comment by msluszniak** [comment]
Yes, I mean there is no info in argsort

**Comment by msluszniak** [comment]
> Also, please provide all intermediate values so we can see any possible errors.
> 
> It's better if you avoid variable name rebinding as well

![image](https://github.com/elixir-nx/nx/assets/56299341/dd09d8c8-f32a-4b29-af29-6ed8edba7351)
So it looks like it slices a sorted tensor but not in descending but in ascending order.

**Comment by msluszniak** [comment]
And it seems that `Nx.take_along_axis` has the same problem

**Comment by polvalente** [comment]
Try slicing with 1..-1//1 please

**Comment by msluszniak** [comment]
Ok, that works

**Comment by polvalente** [comment]
Yeah, that's because the range will default to follow the direction from start to stop with absolute value 1.

In 1..-1 it defaults to be the same as 1..-1//-1 which ends up being translated in Access to 1..(n-1)//-1 I believe.

@josevalim is there a way for us to identify arity-2 vc arity-3 ranges so that we can fix this edge case?

**Comment by josevalim** [comment]
The truth is that, 99% of the cases, they don't want reversal, so I would just say in the error message to do `first..last//1`.

**Comment by polvalente** [comment]
Ah, right. I was actually thinking about stepped slices, but those aren't our problem. We can raise on negative steps for now.
